model/validation.go

```diff
@@ -204,16 +202,433 @@ func (t Tags) Validate() error {
+
+	errs = validateFloatPercent(errs, terminate.Percent, "terminate")
+
+	return
+}
+
+func validateThrottle(throttle *proxyconfig.L4FaultInjection_Throttle) (errs error) {
+
+	errs = validateFloatPercent(errs, throttle.Percent, "throttle")
+
+	if throttle.DownstreamLimitBps < 0 {
+		errs = multierror.Append(errs, fmt.Errorf("downstream_limit_bps invalid"))
+	}
+
+	if throttle.UpstreamLimitBps < 0 {
+		errs = multierror.Append(errs, fmt.Errorf("upstream_limit_bps invalid"))
+	}
+
+	if throttle.GetThrottleAfterSeconds() < 0 {
+		errs = multierror.Append(errs, fmt.Errorf("throttle_after_seconds invalid"))
+	}
+
+	if throttle.GetThrottleAfterBytes() < 0 {
+		errs = multierror.Append(errs, fmt.Errorf("throttle_after_bytes invalid"))
+	}
+
+	// TODO Check DoubleValue throttle.GetThrottleForSeconds()
+
+	return
+}
+
+// ValidateLoadBalancing validates Load Balancing
+func ValidateLoadBalancing(lb *proxyconfig.LoadBalancing) (errs error) {
+
+	// Currently the policy is just a name, and we don't validate it
+
+	return
+}
+
+// ValidateCircuitBreaker validates Circuit Breaker
+func ValidateCircuitBreaker(cb *proxyconfig.CircuitBreaker) (errs error) {
+
+	if simple := cb.GetSimpleCb(); simple != nil {
+		if simple.MaxConnections < 0 {
+			errs = multierror.Append(errs, fmt.Errorf("circuit_breaker max_connections must be in range [0..]"))
+		}
+		if simple.HttpMaxPendingRequests < 0 {
+			errs = multierror.Append(errs, fmt.Errorf("circuit_breaker max_pending_requests must be in range [0..]"))
+		}
+		if simple.HttpMaxRequests < 0 {
+			errs = multierror.Append(errs, fmt.Errorf("circuit_breaker max_requests must be in range [0..]"))
+		}
+		if simple.SleepWindowSeconds < 0 {
+			errs = multierror.Append(errs, fmt.Errorf("circuit_breaker sleep_window_seconds must be in range [0..]"))
+		}
+		if simple.HttpConsecutiveErrors < 0 {
+			errs = multierror.Append(errs, fmt.Errorf("circuit_breaker http_consecutive_errors must be in range [0..]"))
+		}
+		if simple.HttpDetectionIntervalSeconds < 0 {
+			errs = multierror.Append(errs,
+				fmt.Errorf("circuit_breaker http_detection_interval_seconds must be in range [0..]"))
+		}
+		if simple.HttpMaxRequestsPerConnection < 0 {
+			errs = multierror.Append(errs,
+				fmt.Errorf("circuit_breaker http_max_requests_per_connection must be in range [0..]"))
+		}
+		errs = validatePercent(errs, simple.HttpMaxEjectionPercent, "circuit_breaker http_max_ejection_percent")
+	}
+
+	return
+}
+
+func validateWeights(routes []*proxyconfig.DestinationWeight, defaultDestination string) (errs error) {
+
+	// Sum weights by destination
+	weights := map[string]int32{}
+	for _, destWeight := range routes {
+		destination := destWeight.Destination
+		if destination == "" {
+			destination = defaultDestination
+		}
+
+		oldWeight := weights[destination]
+		weights[destination] = oldWeight + destWeight.Weight
+	}
+
+	// From cfg.proto "If there is only [one] destination in a rule, the weight value is assumed to be 100.
```

Comment:
Why do we sum by destination? Isn't the total always 100 no matter where destination goes? Also, please check the case a single weight that is not 100.